### PR TITLE
update reva to include decomposedfs nodes-per-space

### PR DESCRIPTION
**Pull request by butonic**
Run ocis ci for https://github.com/cs3org/reva/pull/2554
- add new role to allow accounts service to `create-space` for metadata storage
- add `create-space` w/ constraint OWN to normal user role

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-7</strong>](https://drone.owncloud.com/owncloud/ocis/9591/28/1) failed. Further test are cancelled...

**Review by fschade — APPROVED** [review]
just some minor and optional findings.

File: graph/pkg/service/v0/drives.go [hunk 1]
```diff
@@ -151,6 +152,22 @@ func (g Graph) GetSingleDrive(w http.ResponseWriter, r *http.Request) {
...
+func canCreateSpace(ctx context.Context, ownPersonalHome bool) bool {
+	s := settingssvc.NewPermissionService("com.owncloud.api.settings", grpc.DefaultClient)
+
+	pr, err := s.GetPermissionByID(ctx, &settingssvc.GetPermissionByIDRequest{
+		PermissionId: settingsServiceExt.CreateSpacePermissionID,
+	})
+	if err != nil || pr.Permission == nil {
+		return false
+	}
+	// TODO @C0rby shouldn't the permissions service check this? aka shouldn't we call CheckPermission?
```
[the remark this answers is not in the document]
Comment (C0rby) [reply graph/pkg/service/v0/drives.go:164]: It's something for later. Yes, ideally we should now replace all permission checks with `CheckPermission` calls to the permissions API.

**Comment by micbar** [comment]
@fschade the failing webUI smoke test smells not good. Something to do with the latest reva change https://github.com/cs3org/reva/pull/2596 ?

File: graph/pkg/service/v0/drives.go [hunk 2]
```diff
@@ -159,12 +176,8 @@ func (g Graph) CreateDrive(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	s := settingssvc.NewPermissionService("com.owncloud.api.settings", grpc.DefaultClient)
-
-	_, err := s.GetPermissionByID(r.Context(), &settingssvc.GetPermissionByIDRequest{
-		PermissionId: settingsServiceExt.CreateSpacePermissionID,
-	})
-	if err != nil {
+	// TODO determine if the user tries to create his own personal space and pass that as a boolean
```
[the remark this answers is not in the document]
Comment (butonic) [reply graph/pkg/service/v0/drives.go:179]: no, it is about the boolean flag in the line below

File: settings/pkg/service/v0/settings.go [hunk 3]
```diff
@@ -366,6 +369,24 @@ func generatePermissionRequests() []*settingssvc.AddSettingToBundleRequest {
...
 		},
+		{
+			BundleId: BundleUUIDRoleUser,
+			Setting: &settingsmsg.Setting{
+				Id:          CreateSpacePermissionID,
+				Name:        CreateSpacePermissionName,
+				DisplayName: "Create own Space",
+				Description: "This permission allows to create a space owned by the current user.",
+				Resource: &settingsmsg.Resource{
+					Type: settingsmsg.Resource_TYPE_SYSTEM, // TODO resource type space? self? me? own?
```
[the remark this answers is not in the document]
Comment (butonic) [reply settings/pkg/service/v0/settings.go:380]: Nothing for this PR. I need to better understand the current permissions and what kind of resource we should put here.

File: settings/pkg/service/v0/settings.go [hunk 4]
```diff
@@ -402,11 +423,35 @@ func generatePermissionRequests() []*settingssvc.AddSettingToBundleRequest {
...
 		},
+		{
+			BundleId: BundleUUIDRoleMetadata,
+			Setting: &settingsmsg.Setting{
+				Id:          CreateSpacePermissionID,
+				Name:        CreateSpacePermissionName,
+				DisplayName: "Create own Space",
+				Description: "This permission allows to create a space owned by the current user.",
+				Resource: &settingsmsg.Resource{
+					Type: settingsmsg.Resource_TYPE_SYSTEM, // TODO resource type space? self? me? own?
```
[the remark this answers is not in the document]
Comment (butonic) [reply settings/pkg/service/v0/settings.go:434]: Nothing for this PR. I need to better understand the current permissions and what kind of resource we should put here.

**Pull request merged**